packages/core/src/components/gesture/gesture.tsx

```diff
@@ -106,29 +123,37 @@ export class Gesture {
     const timeStamp = now(ev);
 
     if (this.lastTouch === 0 || (this.lastTouch + MOUSE_WAIT < timeStamp)) {
-      this.enableMouse(true);
-      this.enableTouch(false);
-
-      this.pointerDown(ev, timeStamp);
+      if (this.pointerDown(ev, timeStamp)) {
+        this.enableMouse(true);
+        this.enableTouch(false);
+      } else {
+        this.abortGesture();
+      }
     }
   }
 
-
   private pointerDown(ev: UIEvent, timeStamp: number): boolean {
-    if (!this.gesture || this.hasStartedPan) {
+    if (!this.gesture || this.hasStartedPan || !this.hasFiredStart) {
+      debugger;
       return false;
     }
-
     const detail = this.detail;
 
-    detail.startX = detail.currentX = pointerCoordX(ev);
-    detail.startY = detail.currentY = pointerCoordY(ev);
+    updateDetail(ev, detail);
+    detail.startX = detail.currentX;
+    detail.startY = detail.currentY;
     detail.startTimeStamp = detail.timeStamp = timeStamp;
     detail.velocityX = detail.velocityY = detail.deltaX = detail.deltaY = 0;
-    detail.directionX = detail.directionY = detail.velocityDirectionX = detail.velocityDirectionY = null;
     detail.event = ev;
     this.positions.length = 0;
 
+    assert(this.hasFiredStart, 'fired start must be false');
+    assert(!this.hasStartedPan, 'pan can be started at this point');
+    assert(!this.hasCapturedPan, 'pan can be started at this point')
+    assert(!this.isMoveQueued, 'some move is still queued');
+    assert(this.positions.length === 0, 'positions must be emprty');
```

Comment:
comment out asserts (or add easy way to remove them when building in production)